badger_voter_sdk/collectors/snapshot_collectors.py

```diff
@@ -42,3 +120,52 @@ def get_voters(snapshot_id: str) -> Optional[Dict]:
         for vote in result['votes']:
             voters[vote['voter']] = vote.get('choice')
     return voters
+
+
+def get_snapshot_by_id(snapshot_id: str) -> Optional[Dict]:
+    """
+    Get single snapshot by id
+    """
+    client = make_gql_client(SNAPSHOT_GQL_API_URL)
+    result = client.execute(GET_SINGLE_PROPOSAL_Q(snapshot_id))
+    if not result or not result.get("proposals"):
+        return
+    target_snapshot = None
+    for proposal in result['proposals']:
+        if proposal['id'] == snapshot_id:
+            target_snapshot = proposal
+    return target_snapshot
+
+
+def get_gauge_weight_snapshot(web3: Web3, space: str) -> Optional[Dict]:
+    """
+    Using title re match and some pool heuristics, tries to get current active
+    gauge voting proposal. If not found, returns None
+    :params:
+    - web3: Web3 instance with valid node url
+    - space: snapshot voting space, like aurafinance.eth
+    """
+    client = make_gql_client(SNAPSHOT_GQL_API_URL)
+    limit = 100
+    offset = 0
+    while True:
+        result = client.execute(GET_ACTIVE_PROPOSALS_Q(first=limit, skip=offset, space=space))
+        offset += limit
+        if not result or not result.get("proposals"):
+            break
```

Comment:
should this raise error? or return obj other than None?

Reply:
Why? It breaks the loop. None is fine, since we are checking that there is no snapshot yet to vote for

Reply:
Ya ok seems fine, makes sense